lib/web/server.cpp

```diff
@@ -0,0 +1,719 @@
+		header->postData = ns->read(header->contentLength);
+		std::cout << header->postData << "\n";
+		mapPostData(header->postMap, header->postData);
+	}
+	//std::cout << "Socialite::Web::Server::readHeader done\n";
+	return true;
+}
+
+string Socialite::Web::Server::cgiLaunch(string file, HttpHeader* header, string getData) {
+	// int pipeFds[2];
+	// if(pipe(pipeFds)) {
+	// 	cout << "failed to create pipe\n";
+	// 	return ""; // failed to create pipe
+	// }
+	// pid_t pid = fork();
+	// if(pid == 0) { // child
+	// 	// swap stdin and stdout for our pipe
+
+	// 	dup2(pipeFds[0], STDIN);
+	// 	close(pipeFds[0]);
+	// 	dup2(pipeFds[1], STDOUT);
+	// 	close(pipeFds[1]);
+
+	// 	User* user = NULL;
+	// 	string passString = "";
+	// 	if(header->cookies["auth-name"] != "" && header->cookies["auth-hash"] != "" && userAccounts->verify(header->cookies["auth-name"], header->cookies["auth-hash"])) {
+	// 		user = userAccounts->getUser(header->cookies["auth-name"]);
+	// 		passString = string(user->username) + ":" + string(user->fName) + ":" + string(user->lName) + ":" + string(user->email);
+	// 	}
+	// 	if(getData == "") {
+	// 		getData = "null";
+	// 	}
+	// 	execl("./PythonBridge", "./PythonBridge", file.c_str(), header->URI.c_str(), header->host.c_str(), getData.c_str(), passString.c_str(), NULL);
+
+	// 	cout << "ERROR: could not launch ./PythonBridge\n";
+	// 	exit(1);
+	// }
+	// else { // parent
+	// 	string ret = "";
+	// 	int c;
+	// 	FILE* stream = fdopen (pipeFds[0], "r");
+	// 	//cout << "getting ready to read\n";
+	// 	while ((c = fgetc(stream)) != 0x0) {
+	// 		ret += c;
+	// 	}
+	// 	fclose(stream);
+	// 	// cleanup
+	// 	int returnValue = waitpid(pid, NULL, 0); // blocking until it's done
+	// 	close(pipeFds[0]);
+	// 	close(pipeFds[1]);
+
+	// 	return ret;
+	// }
+	// return "";
+}
+
+string Socialite::Web::Server::serverAPI(HttpHeader* header, HttpResponseHeader* rHeader, string ip, smatch matcher, bool* success) {
+	*success = false;
+	return string("override");
+}
+
+// the meat of the actual processing
+void Socialite::Web::Server::handleConnection(NetStream* stream) {
+	int error = 0;
+	HttpHeader header;
+	HttpResponseHeader rHeader;
+	struct stat fileStat;
+	HostConfig* currentHost;
+	smatch matcher;
+	int connectionUse = 0;
+
+	stream->setTimeout(5 * 60); // die after 5 minutes of inactivity
+
+	do { // process each request as it comes in if KEEP_ALIVE, otherwise fall through
+		time_t now;
+		time(&now);
+		struct tm* nowTime = gmtime(&now); // shouldn't it be gmtime() who knows...
+		rHeader.lastModified = mktime(nowTime); // default
+		rHeader.size = -1;
+		rHeader.sendContent = true;
+		rHeader.isFile = false;
+		rHeader.transferEncoding = "";
+		rHeader.runCGI = false;
+		rHeader.isForbidden = false;
+		rHeader.getData = "";
+		rHeader.setCookie = "";
+		rHeader.relocate = false;
+		rHeader.allowCache = true;
+
+		bool serverMatched = false;
+		regex apiShape("^/api/([^/]+)/([^/]+)/([^/]+)(?:[/]*)$");
```

Comment:
Don't think this is used anymore since this is coming from the rewrite rules in the server config file now.